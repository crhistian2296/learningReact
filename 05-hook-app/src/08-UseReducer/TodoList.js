import React from 'react';
import TodoItem from './TodoItem';

const TodoList = ({ todosList, handleDelete, handleToggle }) => (
  <div className='col-sm-5'>
    <ol className='list-group list-group-flush'>
      {todosList.map((todo, index) => (
        <TodoItem
          key={index}
          todo={todo}
          index={index}
          handleDelete={handleDelete}
          handleToggle={handleToggle}
        />
      ))}
    </ol>
  </div>
);

export default TodoList;
