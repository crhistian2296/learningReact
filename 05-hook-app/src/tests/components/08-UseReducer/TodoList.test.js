import { shallow } from 'enzyme';
import TodoList from '../../../08-UseReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

const handleDelete = jest.fn();
const handleToggle = jest.fn();
describe('Pruebas en <TodoList/>', () => {
  const wrapper = shallow(
    <TodoList todosList={demoTodos} handleDelete={handleDelete} handleToggle={handleToggle} />
  );
  test('Debe de renderizarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener 2 <TodoItem/>', () => {
    expect(wrapper.find('TodoItem').length).toEqual(demoTodos.length);
    expect(typeof wrapper.find('TodoItem').at(0).prop('handleDelete')).toBe('function');
  });
});
