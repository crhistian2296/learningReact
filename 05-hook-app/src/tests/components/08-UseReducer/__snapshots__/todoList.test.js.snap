// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList/> Debe de renderizarse correctamente 1`] = `
<div
  className="col-sm-5"
>
  <ol
    className="list-group list-group-flush"
  >
    <TodoItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="0"
      todo={
        Object {
          "description": "Aprender Typescript",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="1"
      todo={
        Object {
          "description": "Aprender Next",
          "done": false,
          "id": 2,
        }
      }
    />
  </ol>
</div>
`;
